### PR TITLE
Fix nullpointer when no interest

### PR DESCRIPTION
**Pull request by timja-kainos**

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/InterestDateMapper.java [hunk 1]
```diff
@@ -10,6 +10,10 @@
 
     @Override
     public CCDInterestDate to(InterestDate interestDate) {
+        if (interestDate == null || interestDate.getType() == null) {
+            return null;
```
Comment (rlewan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/InterestDateMapper.java]: Can we return `Optional` here?

Comment (timja-kainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/InterestDateMapper.java]: Not sure if we could get any benefit here, the mappers aren't invoked directly and the pattern has been to return null from there if they could be empty.

Comment (rlewan) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/InterestDateMapper.java]: I had a look into the `Mappper` interface and you're right.

Another matter - `|| interestDate.getType() == null` - it seems to me an interest date without a type is an application error, valid object should be sent to us from frontend. I would prefer not to hid the problem here.

Comment (timja-kainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/InterestDateMapper.java]: not a frontend problem from what I can tell so far, but no idea why its set. Will dig a bit more but would like to merge this if we can't find it soon

**Pull request merged**